WebApplication/ClientApp/src/components/bom.js

```diff
@@ -0,0 +1,137 @@
+/////////////////////////////////////////////////////////////////////
+// Copyright (c) Autodesk, Inc. All rights reserved
+// Written by Forge Design Automation team for Inventor
+//
+// Permission to use, copy, modify, and distribute this software in
+// object code form for any purpose and without fee is hereby granted,
+// provided that the above copyright notice appears in all copies and
+// that both that copyright notice and the limited warranty and
+// restricted rights notice below appear in all supporting
+// documentation.
+//
+// AUTODESK PROVIDES THIS PROGRAM "AS IS" AND WITH ALL FAULTS.
+// AUTODESK SPECIFICALLY DISCLAIMS ANY IMPLIED WARRANTY OF
+// MERCHANTABILITY OR FITNESS FOR A PARTICULAR USE.  AUTODESK, INC.
+// DOES NOT WARRANT THAT THE OPERATION OF THE PROGRAM WILL BE
+// UNINTERRUPTED OR ERROR FREE.
+/////////////////////////////////////////////////////////////////////
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import 'react-base-table/styles.css';
+import './bom.css';
+import { getActiveProject, getBom } from '../reducers/mainReducer';
+import { fetchBom } from '../actions/bomActions';
+import BaseTable, { AutoResizer, Column } from 'react-base-table';
+import styled from 'styled-components';
+
+const Container = styled.div`
+  height: 100vh;
+`;
+
+// compute text width from canvas
+function getMaxColumnTextWidth(strings) {
+  const font = "13px ArtifaktElement, sans-serif";
+  const canvas = document.createElement("canvas");
+  const context2d = canvas.getContext("2d");
+  context2d.font = font;
+  let maxWidth = 0;
+  strings.forEach(element => {
+    const width = context2d.measureText(element).width;
+    maxWidth = width>maxWidth ? width : maxWidth;
+  });
+
+  const roundTo = 10;
+  const rounded = (maxWidth % roundTo==0) ? maxWidth : maxWidth-maxWidth%roundTo + roundTo;
```

Comment:
Could you explain what it does? (maybe it's a thing for a comment)

Reply:
yes, I can put comment here, when text has for example width 81.5, is rounded to 90, 87.1 -> 90, etc